src/Microsoft.Sbom.Parsers.Spdx22SbomParser/Parser/SbomFileParser.cs

```diff
@@ -148,63 +144,41 @@ private void ParseProperty(ref Utf8JsonReader reader, ref byte[] buffer)
         {
             case FileNameProperty:
                 ParserUtils.Read(stream, ref buffer, ref reader);
-                sbomFile.Path = ParseNextString(ref reader, ref buffer);
+                sbomFile.FileName = ParserUtils.ParseNextString(stream, ref reader);
                 break;
 
             case SPDXIdProperty:
                 ParserUtils.Read(stream, ref buffer, ref reader);
-                sbomFile.Id = ParseNextString(ref reader, ref buffer);
+                sbomFile.SPDXId = ParserUtils.ParseNextString(stream, ref reader);
                 break;
 
             case ChecksumsProperty:
                 ParserUtils.Read(stream, ref buffer, ref reader);
-                sbomFile.Checksum = ParseChecksumsArray(ref reader, ref buffer);
+                sbomFile.FileChecksums = ParseChecksumsArray(ref reader, ref buffer);
                 break;
 
             case LicenseConcludedProperty:
                 ParserUtils.Read(stream, ref buffer, ref reader);
-                sbomFile.LicenseConcluded = ParseNextString(ref reader, ref buffer);
+                sbomFile.LicenseConcluded = ParserUtils.ParseNextString(stream, ref reader);
                 break;
 
             case CopyrightTextProperty:
                 ParserUtils.Read(stream, ref buffer, ref reader);
-                sbomFile.FileCopyrightText = ParseNextString(ref reader, ref buffer);
+                sbomFile.FileCopyrightText = ParserUtils.ParseNextString(stream, ref reader);
                 break;
 
             case LicenseInfoInFilesProperty:
                 ParserUtils.Read(stream, ref buffer, ref reader);
-                sbomFile.LicenseInfoInFiles = ParseLicenseInfoInFilesArray(ref reader, ref buffer);
+                sbomFile.LicenseInfoInFiles = ParserUtils.ParseListOfStrings(stream, ref reader, ref buffer);
                 break;
 
             default:
-                SkipProperty(ref reader, ref buffer);
-                break;
-        }
-    }
-
-    private List<string> ParseLicenseInfoInFilesArray(ref Utf8JsonReader reader, ref byte[] buffer)
-    {
-        var licenses = new List<string>();
-
-        // Read the opening [ of the array
-        ParserUtils.AssertTokenType(stream, ref reader, JsonTokenType.StartArray);
-
-        while (reader.TokenType != JsonTokenType.EndArray)
-        {
-            ParserUtils.Read(stream, ref buffer, ref reader);
-            if (reader.TokenType == JsonTokenType.EndArray)
-            {
+                ParserUtils.SkipProperty(stream, ref buffer, ref reader);
                 break;
-            }
-
-            licenses.Add(reader.GetString());
         }
-
-        ParserUtils.AssertTokenType(stream, ref reader, JsonTokenType.EndArray);
-        return licenses;
     }
 
-    private IEnumerable<Checksum> ParseChecksumsArray(ref Utf8JsonReader reader, ref byte[] buffer)
+    private List<Checksum> ParseChecksumsArray(ref Utf8JsonReader reader, ref byte[] buffer)
```

Comment:
Is moving from `IEnumerable` to a `List` necessary here?  I'm just wondering if this might potentially be very large and enumerating it here could be problematic.